Library/Homebrew/cleanup.rb

```diff
@@ -90,27 +99,26 @@ def stale_formula?(scrub)
 
           return false if formula.blank?
 
-          resource_name = basename.to_s[/\A.*?--(.*?)--?(?:#{Regexp.escape(version)})/, 1]
+          resource_name = basename_str[/\A.*?--(.*?)--?(?:#{Regexp.escape(version.to_s)})/, 1]
 
           if resource_name == "patch"
             patch_hashes = formula.stable&.patches&.select(&:external?)&.map(&:resource)&.map(&:version)
             return true unless patch_hashes&.include?(Checksum.new(version.to_s))
           elsif resource_name && (resource_version = formula.stable&.resources&.dig(resource_name)&.version)
             return true if resource_version != version
-          elsif version.is_a?(PkgVersion)
-            return true if formula.pkg_version > version
```

Comment:
Sorbet thinks this branch is unreachable, I believe [this](https://github.com/Homebrew/brew/pull/8378/files#diff-b470fe3421d8ace2adb68e0f68ace29e33cb12f06ebf047ff58a26b579733d33R79) change ensures `version` can only be a `Version` at this point.

Reply:
I agree. It seems like it even if it is an instance of the `PkgVersion` class that it will get turned into an instance of the `Version` class on line 88 after your changes.